### ISSUE TITLE
[Usage]: Obtaining success / error rate % metrics

### ISSUE THREAD

**Issue by yqlu** [vllm-project/vllm#9346]
### Your current environment

Running vLLM v0.5.1 on GKE, but my exact setup isn't relevant to the question below

### How would you like to use vllm

I see that in [vllm/engine/metrics.py](https://github.com/vllm-project/vllm/blob/main/vllm/engine/metrics.py#L144C19-L144C45) there is a success count metric, split up by success reason (anecdotally for me `length` and `stop`). 

Is it currently possible to get a success rate % metric by dividing this by a denominator? What denominator should I use here -- maybe `num_requests_running` or `time_to_first_token_seconds_count`? I tried both, but they didn't seem to provide the right result (in that the ratio could potentially momentarily go above 100). 

If there was a error count metric, I could graph success % as the fraction success / (success + error). Is that in the works? I see [a roadmap](https://github.com/vllm-project/vllm/issues/3616#issuecomment-2030858781) from April which states that `request_failure` is a planned addition, but wasn't sure if this is up to date. Thanks!

### Before submitting a new issue...

- [X] Make sure you already searched for relevant issues, and asked the chatbot living at the bottom right corner of the [documentation page](https://docs.vllm.ai/en/latest/), which can answer lots of frequently asked questions.

**Comment by mces89** [comment]
this metric is quite important for production monitoring, hope to see it soon.

**Comment by mces89** [comment]
@yqlu do you find any solution for this issue? Thanks.

**Comment by yqlu** [comment]
No, I couldn't find any workaround or way to derive this with the existing metrics.

**Comment by achandrasekar** [comment]
@yqlu looks like http metrics is an option here to look for non 2xx status codes - https://docs.vllm.ai/en/latest/design/v1/metrics.html#prometheus-client-library.
